boxoffice/views/order.py

```diff
@@ -0,0 +1,198 @@
+from datetime import datetime
+import decimal
+from flask import url_for, request, jsonify, render_template, abort
+from flask.ext.cors import cross_origin
+from rq import Queue
+from redis import Redis
+from coaster.views import load_models
+from boxoffice import app, ALLOWED_ORIGINS
+from boxoffice.models import db, Organization
+from boxoffice.models import ItemCollection, LineItem, Item, DiscountCoupon, DiscountPolicy
+from boxoffice.models import Order, OnlinePayment, PaymentTransaction, User, PaymentFailError
+from boxoffice.extapi import razorpay
+from forms import LineItemForm, BuyerForm
+from boxoffice.mailclient import send_invoice_email
+from utils import xhr_only, api_result
+
+redis_connection = Redis()
+boxofficeq = Queue('boxoffice', connection=redis_connection)
+
+
+def jsonify_line_items(line_items):
+    """
+    Serializes and return line items in the format:
+    {item_id: {'quantity': Y, 'final_amount': Z, 'discounted_amount': Z, 'discount_policy_ids': ['d1', 'd2']}}
+    """
+    items_json = dict()
+    for line_item in line_items:
+        if not items_json.get(unicode(line_item.item_id)):
+            items_json[unicode(line_item.item_id)] = {'quantity': 0, 'final_amount': decimal.Decimal(0), 'discounted_amount': decimal.Decimal(0), 'discount_policy_ids': []}
+        if not items_json[unicode(line_item.item_id)].get('final_amount'):
+            items_json[unicode(line_item.item_id)]['final_amount'] = decimal.Decimal(0)
+        items_json[unicode(line_item.item_id)]['final_amount'] += line_item.base_amount - line_item.discounted_amount
+        items_json[unicode(line_item.item_id)]['discounted_amount'] += line_item.discounted_amount
+        items_json[unicode(line_item.item_id)]['quantity'] += 1
+        if line_item.discount_policy_id and line_item.discount_policy_id not in items_json[unicode(line_item.item_id)]['discount_policy_ids']:
+            items_json[unicode(line_item.item_id)]['discount_policy_ids'].append(line_item.discount_policy_id)
+    return items_json
+
+
+@app.route('/kharcha', methods=['OPTIONS', 'POST'])
```

Comment:
This will conflict with an organization named `kharcha`. Move one out of the way.